src/compiler/crystal/macros.cr

```diff
@@ -1187,6 +1187,17 @@ module Crystal::Macros
   class Out < UnaryExpression
   end
 
+  # An `offsetof` expression.
+  class OffsetOf < ASTNode
+    # Returns the type that has been used in this `offsetof` expression.
+    def structure : TypeNode
```

Comment:
I'm not sure about the naming if classes are allowed.

Reply:
Why not simply call it `type`?

Reply:
We can call it `type` in the macros module, but its internal name must be something else since `ASTNode` already defines `@type`. Maybe `base`, like "we take the offset of an element from its base"?

Reply:
Right. I'd simply go with `offset_type` then. No need to introduce another name.